python/cudf/cudf/core/buffer.py

```diff
@@ -5,108 +5,206 @@
 import functools
 import operator
 import pickle
-from typing import Any, Dict, Tuple
+from typing import (
+    Any,
+    Dict,
+    Mapping,
+    Protocol,
+    Tuple,
+    Union,
+    runtime_checkable,
+)
 
 import numpy as np
 
 import rmm
 
 import cudf
 from cudf.core.abc import Serializable
+from cudf.utils.string import format_bytes
+
+
+@runtime_checkable
+class DeviceBufferLike(Protocol):
```

Comment:
I'm worried about the naming here; it's a bit confusing that `rmm.DeviceBuffer` is not a `cudf.DeviceBufferLike`.

At a high level this is also related to https://github.com/rapidsai/rmm/issues/227; I think we can leave the name as is for now and just wait until `DeviceBufferLike` and `Buffer` are moved to RMM and/or reconciled with `rmm.DeviceBuffer`.

Reply:
Agree